core/src/main/java/org/elasticsearch/rest/action/RestToXContentListener.java

```diff
@@ -41,22 +44,19 @@ public final RestResponse buildResponse(Response response) throws Exception {
         return buildResponse(response, channel.newBuilder());
     }
 
-    public final RestResponse buildResponse(Response response, XContentBuilder builder) throws Exception {
-        if (wrapInObject()) {
+    protected final void toXContent(Response response, XContentBuilder builder) throws IOException {
+        if (response instanceof ToXContentObject == false) {
```

Comment:
It'd be nice to be able to change this to only take `ToXContentObject` one day....